Add documentation for Advanced Traffic Management (#1221)
humanmade/altis-cloud: docs/WAF/traffic-mangement.md

# Traffic Management

The **Traffic Management** feature gives administrators self-service control over the automated traffic — bots,
crawlers, scrapers, and scripted clients — that reaches your environment. Rather than
maintaining lists of individual user agents, you define rules based on traffic *categories*
(such as AI crawlers or SEO tools) and *signals* (such as requests originating from known data centers).

> For each type of traffic, you can choose whether to allow it, block it, or require it to prove that it's human.

Reviewer: ```suggestion
The Traffic Management feature gives administrators self-service control over the automated traffic — bots,
crawlers, scrapers, and scripted clients — that reaches your environment. Rather than
maintaining lists of individual user agents, you define rules based on traffic categories
(such as AI crawlers or SEO tools) and signals (such as requests originating from known data centers).
For each type of traffic, you can choose whether to allow it, block it, or require it to prove that it's human.
```

Traffic Management extends the default [Web Application Firewall](./README.md) protections available to
all Altis customers with the detection and control capabilities provided by AWS WAF Bot Control. Altis
handles the underlying setup and surfaces AWS WAF detection results in plain language through the
Dashboard, giving you self-service control over automated traffic without requiring any code or
configuration changes. You decide what to allow, block, or challenge.

> 

Reviewer: I would drop this entirely; I think it's repetitive, and once we ship this, it doesn't *extend* our WAF, it's just part of it.

```suggestion
```

More advanced detections, such as silent background Challenge actions and machine-learning-based
targeted rules, require the Advanced Traffic Management add-on. See [What the Advanced Traffic
Management add-on adds](#what-the-advanced-traffic-management-add-on-adds) for more details.

For simpler, HTTP header-based control, see [User-Agent Blocking](./ua-blocking.md). To manage

> access at the IP address level, see [IP Access Control](./access-control.md).

Reviewer: ```suggestion
For precise control per user-agent, see [User-Agent Blocking](./ua-blocking.md). To manage
access by IP address or range, see [IP Access Control](./access-control.md).
```

## What Traffic Management does

Every request to your environment is inspected and classified using a range of signals —
user-agent strings, IP address reputation, request patterns, and (for the add-on) browser

> interrogation and machine-learning analysis. Traffic Management lets you:

Reviewer: ```suggestion
Altis automatically inspects and classifies your traffic using a range of
signals, including user-agents, IP addresses, and request patterns.

Traffic Management lets you:
```

- **See** how much of your traffic is automated, broken down by category and signal.
  With the [add-on](#what-the-advanced-traffic-management-add-on-adds), you can also see individual
  bot names and the organizations behind them.
- **Decide** how each type of automated traffic is handled: allow it, block it, or make it
  prove it's human.

Traffic Management appears in two places in the Dashboard:

- The **Traffic** page shows analytics: how your traffic breaks down between verified bots,
  unverified bots, and non-bot (human) requests, and what has been allowed or blocked.
- The **Traffic Management** settings page (under **Application Settings**) is where you

>   configure the mode and choose an action for each rule.

Reviewer: ```suggestion
- **Environment → Settings → Traffic Management** is where you configure the
  mode and choose an action for each rule.
```

| **Allow**     | Matching requests are permitted through. Use this for traffic you want to keep, such as search engines.                                                                                                                                                                                                                            |
| **Block**     | Matching requests are rejected with an HTTP `403` and never reach your application.                                                                                                                                                                                                                                                |
| **CAPTCHA**   | Matching requests are shown a CAPTCHA puzzle on an interstitial page. Humans can solve it and continue; most bots cannot. Solving it issues a token so the visitor isn't repeatedly challenged.                                                                                                                                    |

> | **Challenge** | Matching requests are given a silent, background browser check — no puzzle is shown. Real browsers pass automatically and invisibly; automated clients that can't run the challenge are stopped. **Challenge is available only with the [Advanced Traffic Management add-on](#what-the-advanced-traffic-management-add-on-adds).** |

Reviewer: ```suggestion
| Action                       | What happens                                                                                                                                                                                     |
|------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| **Allow**                    | Matching requests are permitted through. Use this for traffic you want to keep, such as search engines.                                                                                          |
| **Block**                    | Matching requests are rejected with an HTTP `403` and never reach your application.                                                                                                              |
| **CAPTCHA**                  | Matching requests are shown a CAPTCHA puzzle on an interstitial page. Humans can solve it and continue; most bots cannot. Solving it issues a token so the visitor isn't repeatedly challenged.  |
| **Challenge** (requires ATM) | Matching requests are given a silent, background browser check — no puzzle is shown. Real browsers pass automatically and invisibly; automated clients that can't run the challenge are stopped. |
```

The CAPTCHA puzzle is accessible — it offers both visual and audio variants and is available
in multiple languages. The interstitial page shown while a CAPTCHA is pending **does not count
towards your page views**.

> 

Reviewer: ```suggestion
Blocked requests and requests which do not pass the CAPTCHA or Challenge tests are not counted
towards your page views. (The interstitial page for CAPTCHAs and Challenges are also not
counted as views.)

The CAPTCHA puzzle is accessible — it offers both visual and audio variants and is available
in multiple languages. CAPTCHAs are only displayed to the user on their first visit, and
a token is stored in their cookies to ensure they don't see it again.
```

> **CAPTCHA vs. Challenge.** A CAPTCHA interrupts the visitor with a puzzle, so use it where
> some friction is acceptable, or where a rule might occasionally misidentify a real user and
> you'd rather give them a way through than block them outright. A Challenge is invisible to
> real users, making it the better default for protecting normal pages — if you have the
> [Advanced Traffic Management add-on](#what-the-advanced-traffic-management-add-on-adds).

## Rule groups

The settings page splits the rules into three groups.

### Bot Categories

Bots that identify themselves are grouped into categories based their purpose. Blocking or challenging a category
affects only unverified bots in it — verified bots like Googlebot are left alone. AI is the one

> exception (see below).

Reviewer: This isn't true in our implementation, it applies to all.

```suggestion
Bots are automatically classified into into categories based their purpose, as
determined by self-reported user-agent and other signals.
```

> | Category           | Description                                    |

Reviewer: I wonder if we should drop this explicit list entirely in favour of the one in the Dashboard itself; I can see this falling out of date.

| Archiver           | Web archiving bots                             |
| Content Fetcher    | Content fetching services (e.g. RSS readers)   |
| Email Client       | Email client link previews                     |
| HTTP Library       | HTTP client libraries used by scripts and APIs |
| Link Checker       | Link validation bots                           |
| Miscellaneous      | Uncategorized bots                             |
| Monitoring         | Uptime and monitoring services                 |
| Scraping Framework | Web scraping frameworks                        |
| Search Engine      | Search engine crawlers                         |
| Security           | Security scanners and crawlers                 |
| SEO                | SEO analysis tools                             |
| Social Media       | Social media crawlers and link-preview embeds  |
| AI                 | AI training and inference crawlers             |

> **The AI category works differently.** Unlike other categories, the action you choose for the AI category applies
> to *all* matching bots, verified or not. This lets you block AI crawlers even when
> they identify themselves honestly.

> 

Reviewer: Per above.
```suggestion
```

### Signals

Signals are indicators that a request is automated, these are used for traffic that doesn't fall into a clean category.

| Signal                 | Description                                                                          |
|------------------------|--------------------------------------------------------------------------------------|
| Automated Browser      | The client browser shows indicators of being automated (e.g. headless or scripted).  |
| Known Bot Data Center  | The request originates from a hosting provider or data center commonly used by bots. |
| Non-Browser User Agent | The user-agent string doesn't look like a real web browser (often API clients).      |

### Targeted Protections

The most sophisticated detections, aimed at bots that deliberately hide and don't identify
themselves. These use browser checks, fingerprinting, request-rate analysis, and

> machine learning. **Targeted Protections require

Reviewer: ```suggestion
machine learning (ML). **Targeted Protections require
```

the [Advanced Traffic Management add-on.](#what-the-advanced-traffic-management-add-on-adds)**

They fall into a few families:

- **Volumetric** — abnormally high request volumes from a single IP or session in a short
  window (including sessions with no valid token).
- **Token-based** — requests missing a valid session token, which most real browsers carry.
- **Targeted signals** — token-level evidence of automation: automated browsers, browser
  automation extensions (such as Selenium), and inconsistent browser fingerprints.
- **Coordinated activity (ML)** — machine-learning detection of distributed, coordinated bot
  campaigns spread across many IP addresses, reported at Low, Medium, and High confidence.
- **Token reuse (ML)** — a single token being reused across many distinct IP addresses,
  countries, or networks (ASNs), reported at Low, Medium, and High confidence.

Because these rules escalate by confidence level, a common pattern is to Challenge or CAPTCHA

> the lower-confidence levels and Block only the high-confidence ones.

Reviewer: ```suggestion

With the ML-based rules, a confidence rating is assigned to the signal based on
the model's output. A common pattern is to Challenge or CAPTCHA lower-confidence
levels and Block only the high-confidence ones.
```

blocking.

## What the Advanced Traffic Management add-on adds

Every Altis customer gets bot categories and signals with the Allow, Block, and CAPTCHA actions.
The **Advanced Traffic Management add-on** adds:

- **Targeted Protections** — the machine-learning and fingerprint-based rules above.
- **The Challenge action** — silent background browser checks, so you can stop bots without
  showing anyone a puzzle.
- **Richer analytics** — the Identified bots and Bot organizations tables on the Traffic page,
  naming the specific bots hitting your site and the companies behind them.

To enable the add-on, or for help deciding whether it's right for your site, contact your
account manager or Altis support.

## Best practices

- **Start in Count Only.** Enable Traffic Management in Count Only mode and watch the Traffic
  page before enforcing anything, so you can see what would be affected.
- **Protect verified bots you rely on.** Leave categories like Search Engine on Allow so you
  don't harm SEO. Remember verified bots are generally left alone regardless.
- **Prefer Challenge over Block for page traffic.** A background Challenge stops bots without

>   inconveniencing real visitors; reserve Block for traffic you're confident is unwanted.

Reviewer: ```suggestion
- **Prefer CAPTCHA over Block when uncertain.** The CAPTCHA action stops bots while
  still allowing real-users to access the site; reserve Block for traffic you're
  confident is unwanted.
```
